source/2015-11-27-colorize-svgs-with-sass.html.markdown

```diff
@@ -0,0 +1,83 @@
+---
+title: Colorize SVGs with SASS
+date: 2015-11-27 16:05 UTC
+author: bro
+tags: 
+---
+
+As you might know, it is possible to style an SVG image via CSS if you
+inline it. The problem, though, is that you cannot do this when the SVG
+image is used as a background image in CSS.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<svg xmlns="http://www.w3.org/2000/svg" width="400" height="400">
+  <circle cx="100" cy="100" r="50" stroke="black" stroke-width="5" fill="none" />
+</svg>
+```
+
+```css
+.circle {
+  background-image url(circle.svg);
+}
+```
+
+You can use [CSS masks](http://www.w3.org/TR/css-masking/), but not all
+browsers [support it](http://caniuse.com/#feat=css-masks).
+
+```css
+.circle {
+  background-color: blue;
+  mask-image: url(circle.svg);
+}
+```
+
+But if you use SASS, you can write your own SASS function to inline the SVG
+and for example change the color of all strokes (script inspired by 
+[this](https://github.com/Compass/compass/issues/1460#issuecomment-65132746)
+Compass issue):
+
+```ruby
+require 'sass'
+require 'cgi'
+
+module Sass::Script::Functions
+  def colorize_svg(svg_name, color)
```

Comment:
maybe `colorize_svg_strokes` as it only colors the strokes (not sure though...).